docs/content/TreeView.mdx

```diff
@@ -104,6 +108,92 @@ description: A hierarchical list of items where nested items can be expanded and
 </Box>
 ```
 
+### With React Router
+
+```jsx
+import {Link, useMatch, useResolvedPath, navigate} from 'react-router-dom'
+import {TreeView} from '@primer/react'
+
+function LinkItem({id, to, children}) {
+  const resolved = useResolvedPath(to)
+  const isCurrent = useMatch({path: resolved.pathname, end: true})
+  return (
+    <TreeList.Item
+      id={id}
+      aria-current={isCurrent ? 'page' : undefined}
+      current={isCurrent}
+      onSelect={() => navigate(to)}
+    >
+      {children}
+    </TreeList.Item>
+  )
+}
+
+function App() {
+  return (
+    <Box sx={{display: 'grid', gap: 2, maxWidth: 400}}>
```

Comment:
This wrapping Box is probably not necessary for this example. The same applies to the Next.js example.
```suggestion
```